docroot/modules/humsci/hs_courses/hs_courses.module

```diff
@@ -13,3 +16,24 @@ function hs_courses_course_collections_view_alter(array &$build, Drupal\Core\Ent
     unset($build['title']);
   }
 }
+
+/**
+ * Implements hook_search_api_db_query_alter().
+ */
+function hs_courses_search_api_db_query_alter(SelectInterface &$db_query, QueryInterface $query) {
+  $courses = \Drupal::database()
+    ->select('node_field_data', 'n')
+    ->condition('type', 'hs_course')
+    ->fields('n', ['nid', 'title'])
+    ->orderBy('nid', 'asc')
+    ->execute();
+
+  $course_titles = [];
+  // Courses that have multiple sections, we want to exclude the
```

Comment:
Comment left incomplete. I know you are importing multiple course nodes but other developers may not. Please elaborate that you are doing this to prevent duplicate results. 

Also, could this have been done by aggregating or filtering out one ID in the view?

Reply:
Oh wait, nope, cuz different nids and guids. Ignore part about view aggregation. Perhaps maybe on title but that is risky to aggregate on strings that are not forced to be unique.

Reply:
So the views don't even have aggregation options. I wasn't 100% sure about title or if I should do a field query instead. I could then filter out nodes with matching course ids

Reply:
Or I could query the migrate map table. ??

Reply:
I found another hook that is actually the more accurate one to use. it prevents courses from being indexed entirely.

Reply:
https://github.com/SU-HSDO/suhumsci/pull/56/files#diff-06c289cc397560e45f2a5e31f39b57f2R21